### PR TITLE
feature: expose tunable parameters to support multiple tfs

### PR DESCRIPTION
**Pull request by jinpengqi**
Implement new container environment which could expose additional
parameters to help user optimize for throughput and fine tune
container performance.

Co-authored-by: Liang Ma <liangmaa@amazon.com>

*Issue #, if available:*

*Description of changes:*

To elaborate, user could tune environment parameters such as
SAGEMAKER_GUNICORN_WORKERS, SAGEMAKER_TFS_INSTANCE_COUNT,
SAGEMAKER_TFS_INTER_OP_PARALLELISM,
SAGEMAKER_TFS_INTRA_OP_PARALLELISM and etc.
In this way container will be started with multiple of gunicorn
workers and they are responsible for serving requests to different
tensorflow model servers parallelly. Thus, throughput will be
improved based on different parameters tuned.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker/build_artifacts/sagemaker/tfs_utils.py [hunk 1]
```diff
@@ -26,11 +26,12 @@
...
 
 Context = namedtuple("Context",
-                     "model_name, model_version, method, rest_uri, grpc_port, "
+                     "model_name, model_version, method, rest_uri, grpc_port, channel, "
                      "custom_attributes, request_content_type, accept_header, content_length")
 
 
-def parse_request(req, rest_port, grpc_port, default_model_name, model_name=None):
+
+def parse_request(req, rest_port, grpc_port, default_model_name, model_name=None, channel=None):
```
Comment (liangma8712) [inline docker/build_artifacts/sagemaker/tfs_utils.py:34]: can we can place the "channel" arg before "model_name" arg and remove the default None value? I think channel object is always passed to this method.

**Review by liangma8712 — COMMENTED** [review]
Looks good overall. some nit comments.

File: docker/build_artifacts/sagemaker/tfs_utils.py (hunk 1 above)
Comment (jinpengqi) [reply docker/build_artifacts/sagemaker/tfs_utils.py:34]: This method is also called [here](https://github.com/aws/sagemaker-tensorflow-serving-container/blob/master/docker/build_artifacts/sagemaker/python_service.py#L240), where channel is not passed in.
So we need to specify the passed-in channel/model arg while calling this method.

Comment (liangma8712) [reply docker/build_artifacts/sagemaker/tfs_utils.py:34]: I see. make sense.

**Pull request closed without merging**